# generated by patch-package 6.4.8 on 2022-03-12 22:57:59
#
# command:
#   npx patch-package @tarojs/plugin-framework-vue3
#
# declared package:
#   @tarojs/plugin-framework-vue3: 3.4.3
#
diff --git a/node_modules/@tarojs/plugin-framework-vue3/dist/runtime.js b/node_modules/@tarojs/plugin-framework-vue3/dist/runtime.js
index cecd4d6..2962c28 100644
--- a/node_modules/@tarojs/plugin-framework-vue3/dist/runtime.js
+++ b/node_modules/@tarojs/plugin-framework-vue3/dist/runtime.js
@@ -73,6 +73,10 @@ const setGlobalDataPlugin = {
     }
 };
 
+function isClassComponent (value) {
+    return isFunction(value) && '__vccOpts' in value
+}
+
 /**
  * set writable, enumerable to true
  */
@@ -127,6 +131,7 @@ function setReconciler() {
 function createVue3Page(h, id) {
     return function (component) {
         var _a;
+        component = isClassComponent(component) ? component.__vccOpts : component
         const inject = {
             props: {
                 tid: String
@@ -177,7 +182,7 @@ function createVue3Page(h, id) {
 function createVue3App(app, h, config) {
     let pages = [];
     let appInstance;
-    ensure(!isFunction(app._component), '入口组件不支持使用函数式组件');
+    ensure(!(isFunction(app._component) && !isClassComponent(app._component)), '入口组件不支持使用函数式组件');
     setReconciler();
     app._component.render = function () {
         return pages.slice();
